src/cis/fam_cis_direct.cpp

```diff
@@ -850,11 +884,10 @@ void *Fam_CIS_Direct::copy(uint64_t srcRegionId, uint64_t srcOffset,
 
     Fam_Memory_Service *memoryService = get_memory_service(memoryServerId);
 
-    Fam_Metadata_Service *metadataService =
-        get_metadata_service(memoryServerId);
+    Fam_Metadata_Service *metadataService = get_metadata_service(0);
 
-    uint64_t srcDataitemId = srcOffset / MIN_OBJ_SIZE;
-    uint64_t destDataitemId = destOffset / MIN_OBJ_SIZE;
+    uint64_t srcDataitemId = (memoryServerId << 32) + srcOffset / MIN_OBJ_SIZE;
```

Comment:
use get_dataitem_id, BTW is it possible to make get_dataitem_id a macro or inline function.